src/clone/Cloner.php

```diff
@@ -0,0 +1,274 @@
+
+		$site = new EE\Site\Cloner\Site( $sitename, $host, $user, $ssh, [] );
+
+		return $site;
+	}
+
+	public function execute( string $command ): EE\ProcessRun {
+		return EE::launch( $this->get_ssh_command( $command ) );
+	}
+
+	private function get_ssh_command( string $command ): string {
+		return $this->ssh_string ? 'ssh -t ' . $this->ssh_string . ' "' . $command . '"' : $command;
+	}
+
+	public function get_rsync_path( string $path ): string {
+		return $this->ssh_string ? $this->ssh_string . ':' . $path : $path;
+	}
+
+	public function get_site_root_dir(): string {
+		return $this->get_rsync_path( $this->site_details['site_fs_path'] . '/app/htdocs/' );
+	}
+
+	public function validate_ee_version(): void {
+		$result  = $this->execute( 'ee cli version' );
+		$matches = null;
+
+		preg_match( "/^EE (?'version'\S+)/", $result->stdout, $matches );
+		$matches['version'] = preg_replace('/-nightly.*$/', '', $matches['version']);
+	}
+
+	private function get_ssl_args( Site $source_site, $assoc_args ): string {
+		$site_details = $source_site->site_details;
+		$ssl_args     = '';
+		$add_wildcard = false;
+
+		if ( $assoc_args['ssl'] ?? false ) {
+			if ( $assoc_args['ssl'] !== 'off' ) {
+				$ssl_args .= ' --ssl=' . $assoc_args['ssl'];
+				if ( $assoc_args['ssl'] === 'custom' ) {
+					if ( ! ( $assoc_args['ssl-key'] ?? false && $assoc_args['ssl-crt'] ?? false ) ) {
+						throw new \Exception( 'You need to specify --ssl-crt and --ssl-key with --ssl=custom' );
+					}
+					if ( ! is_file( $assoc_args['ssl-crt'] ) ) {
+						throw new \Exception( 'Unable to find file specified in --ssl-crt at \'' . $assoc_args['ssl-crt'] . '\'' );
+					}
+					if ( ! is_file( $assoc_args['ssl-key'] ) ) {
+						throw new \Exception( 'Unable to find file specified in --ssl-key at \'' . $assoc_args['ssl-key'] . '\'' );
+					}
+
+					$rsync_command_crt = rsync_command( $assoc_args['ssl-crt'], $this->get_rsync_path( get_temp_dir() ) );
+					$rsync_command_key = rsync_command( $assoc_args['ssl-key'], $this->get_rsync_path( get_temp_dir() ) );
+
+					if ( ! ( EE::exec( $rsync_command_key ) && EE::exec( $rsync_command_crt ) ) ) {
+						throw new \Exception( 'Unable to sync certs.' );
+					}
+
+					$ssl_args .= ' --ssl-crt=\'' . get_temp_dir() . basename( $assoc_args['ssl-crt'] ) . '\'';
+					$ssl_args .= ' --ssl-key=\'' . get_temp_dir() . basename( $assoc_args['ssl-key'] ) . '\'';
+				}
+				if ( $assoc_args['wildcard'] ?? false ) {
+					$ssl_args .= ' --wildcard';
+				}
+			}
+
+			return $ssl_args;
+		}
+
+		if ( ! empty( $site_details['site_ssl'] ) ) {
+			// If name of src and dest site are same
+			if ( $this->name === $source_site->name ) {
+				if ( $site_details['site_ssl'] === 'le' || $site_details['site_ssl'] === 'custom' ) {
+					EE\Site\Cloner\Utils\copy_site_certs( $source_site, $this );
+					$ssl_args .= ' --ssl=custom --ssl-key=\'' . get_temp_dir() . $source_site->name . '.key\' --ssl-crt=\'' . get_temp_dir() . $source_site->name . '.crt\'';
```

Comment:
`--ssl=custom` this should not be custom always. It should be `--ssl=$site_details['site_ssl']`
As in case of type `le`, the renewal would stop if `custom` is hardcoded.